lib/kastlex/offsets_cache.ex

```diff
@@ -76,28 +95,29 @@ defmodule Kastlex.OffsetsCache do
                          end)
     request_fields = [replica_id: -1, topics: topics]
     request = :kpro.make_request(:list_offsets, 0, request_fields)
-    endpoints = [{String.to_charlist(leader.host), leader.port}]
-    do_fun = fn(connection) ->
-      {:ok, response} = :kpro.request_sync(connection, request, 10_000)
-      responses = kpro_rsp(response)[:msg][:responses]
-      Enum.each(responses,
-                fn(tr) ->
-                  Enum.each(tr[:partition_responses],
-                            fn(pr) ->
-                              try do
-                                [offset] = pr[:offsets]
-                                :ets.insert(@table, {{tr[:topic], pr[:partition]}, offset})
+    leader = Kastlex.MetadataCache.get_leader(id)
+    client_id = Kastlex.get_brod_client_id()
+    {:ok, conn} = :brod_client.get_connection(client_id, String.to_charlist(leader.host), leader.port)
```

Comment:
this is pretty easy to crash if a node is taken down but the metadata is not refreshed yet.
perhaps a better approach is to spawn a worker for each leader and make hem run independently with a `timer:send_interval` (and ignore `{:error, reason}` too).